test/e2e/case9_templates_test.go

```diff
@@ -517,4 +529,308 @@ var _ = Describe("Test policy templates", func() {
+			utils.Kubectl("-n", testNamespace, "create", "sa", "case9-sa-does-not-exist", "--kubeconfig="+kubeconfigHub)
+
+			By("Verifying the replicated policy has the correct values")
+			for i := 0; i < 3; i++ {
+				cluster := fmt.Sprintf("managed%d", i+1)
+
+				By("Checking the policy for " + cluster)
+
+				Eventually(func(g Gomega) {
+					plc := utils.GetWithTimeout(
+						clientHubDynamic,
+						gvrPolicy,
+						testNamespace+".case9-sa-policy",
+						cluster,
+						true,
+						defaultTimeoutSeconds,
+					)
+					g.Expect(plc).NotTo(BeNil())
+
+					tmpls, _, _ := unstructured.NestedSlice(plc.Object, "spec", "policy-templates")
+					g.Expect(tmpls).To(HaveLen(1))
+
+					tmplAnnotations, _, _ := unstructured.NestedStringMap(tmpls[0].(map[string]interface{}),
+						"objectDefinition", "metadata", "annotations")
+					hubTmplErrAnnotation := tmplAnnotations["policy.open-cluster-management.io/hub-templates-error"]
+					g.Expect(hubTmplErrAnnotation).To(BeEmpty())
+
+					tmplLabels, _, _ := unstructured.NestedStringMap(
+						tmpls[0].(map[string]interface{}), "objectDefinition", "metadata", "labels",
+					)
+					g.Expect(tmplLabels).ToNot(BeEmpty())
+					g.Expect(tmplLabels["configmap"]).To(Equal("Raleigh"))
+					g.Expect(tmplLabels["namespace"]).To(Equal("default"))
+				}, defaultTimeoutSeconds, 1).Should(Succeed())
+			}
+		})
+	})
+
+	Describe("Test token issuance and refresh", Ordered, func() {
+		BeforeAll(func(ctx SpecContext) {
+			utils.Kubectl("apply", "-f", case9SATokenYaml, "--kubeconfig="+kubeconfigHub)
+
+			DeferCleanup(func() {
+				utils.Kubectl("delete", "-f", case9SATokenYaml, "--kubeconfig="+kubeconfigHub, "--ignore-not-found")
+			})
+		})
+
+		It("A token should be refreshed within a minute", func(ctx SpecContext) {
+			tokenCtx, tokenCtxCancel := context.WithCancel(ctx)
+
+			defer tokenCtxCancel()
+
+			// Refresh the token between 1 minute and 30 seconds after issuance
+			refreshConfig := propagator.TokenRefreshConfig{
+				ExpirationSeconds: 600, // 10 minutes
+				MinRefreshMins:    9,
+				MaxRefreshMins:    9.5,
+			}
+
+			var tokenPath string
+
+			Eventually(func(g Gomega) {
+				var err error
+
+				tokenPath, err = propagator.GetToken(
+					tokenCtx,
+					&sync.WaitGroup{},
+					clientHub.(*kubernetes.Clientset),
+					types.NamespacedName{Namespace: "default", Name: "test-user"},
+					refreshConfig,
+				)
+				g.Expect(err).ToNot(HaveOccurred())
+			}, defaultTimeoutSeconds, 1).Should(Succeed())
+
+			token, err := os.ReadFile(tokenPath)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(token).ToNot(BeNil())
+
+			By("Monitoring the token file for a change: " + tokenPath)
+
+			Eventually(func(g Gomega) {
+				refreshedToken, err := os.ReadFile(tokenPath)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(refreshedToken).ToNot(BeNil())
+				g.Expect(bytes.Equal(token, refreshedToken)).To(BeFalse(), "Expected the token to have been refreshed")
+			}, 70, 3).Should(Succeed())
```

Comment:
Why 70 not 60?

Reply:
Just because it could be exactly 60 seconds before the token refreshes, so I gave it a bit of a buffer. I could have made it 65 seconds though.

Reply:
I don't think it is a big deal as long as they change it. I don't think no one complain that it is over 1min